src/common/log/parser.py

```diff
@@ -192,19 +193,53 @@ def parse_fmu_xml_log(filename, modulename = 'Model', accept_errors=False):
             print('Parsed log will be incomplete')
         else:
             raise Exception('Failed to parse XML FMU log:\n' + repr(e))
-    
+
     return handler.get_root()
 
-def extract_xml_log(destfilename, filename, modulename = 'Model'):
+def extract_xml_log(dest, log, modulename = 'Model'):
     """
-    Extract the XML contents of a FMU log and write as a new file destfilename.
-
-    modulename selects the module as recorded in the beginning of each line by
+    Extract the XML contents of a FMU log and write as a new file dest, or extract into a stream.
+    Input argument 'modulename' selects the module as recorded in the beginning of each line by
     FMI Library.
+
+        dest::
+
+            file_name  --
+                Name of the file which holds the extracted log, or a stream to write to
+                that supports the function 'write'. Default behaviour is to write to a file.
+                Default: get_log_filename() + xml
+            log --
+                String of filename to extract log from or a stream. The support for stream
+                requires that the stream is readable and supports 'seek' and 'readlines'.
+                Default: get_log_filename() + xml
+            modulename --
+                Selects the module as recorded in the beginning of each line by FMI Library.
+                Default: 'Model'
     """
-    with open(filename, 'r') as sourcefile:
-        with open(destfilename, 'w') as destfile:
-            filter_xml_log(destfile.write, sourcefile, modulename)
+    # if it is a string, we assume we write to a file (since the file doesnt exist yet)
+    dest_is_file = isinstance(dest, str)
+    if not dest_is_file:
+        if not hasattr(dest, 'write'):
+            raise FMUException("If input argument 'dest' is a stream it needs to support the attribute 'write'.")
+
+    if isinstance(log, str):
+        with open(log, 'r') as sourcefile:
+            if dest_is_file:
+                with open(dest, 'w') as destfile:
+                    filter_xml_log(destfile.write, sourcefile, modulename)
+            else:
+                filter_xml_log(dest.write, sourcefile, modulename)
+    elif hasattr(log, 'seek') and hasattr(log, 'readlines'):
+            log.seek(0) # Return to the start of the stream
+            if dest_is_file:
+                with open(dest, 'w') as destfile:
+                    filter_xml_log(destfile.write, log.readlines(), modulename)
+            else:
+                filter_xml_log(dest.write, log.readlines(), modulename)
+    else:
```

Comment:
Why? I think we should try to assume that the stream is at the start. We should not assume to need to seek in the stream.

Reply:
I would be OK with that but that would also mean that the following workflow is not going to work:
```
fmu.simulate()
fmu.extract_xml_log()
```
Because the user would have to reset the stream postion in-between, wouldn't it be valuable to keep this? Or what do you think?

Reply:
I agree that makes sense:
See https://github.com/modelon-community/PyFMI/commit/8e355093157fb6e9b1824156bc923aa8a873cb86